File: src/test/java/AdvisorTest.java
import iteration2.*;
import org.junit.Before;
import org.junit.Test;
import static org.junit.Assert.*;

import java.io.IOException;
import java.util.ArrayList;

public class AdvisorTest {
    private Advisor[] advisors = {new Advisor("advisor")};
    private Student student;
    private Curriculum curriculum = new Curriculum();
    private JsonParser parser = new JsonParser();
    private SystemParameter parameters = new SystemParameter(1,70,10,10,40);


    @Before
    public void initialize() throws IOException {
        student = new Student("Sanji","1501210017",1,new Transcript());
        student.setAdvisor(advisors[0]);
        parser.parseCourseObjects(curriculum,advisors);
        parameters.setMaxCoursePerSemester(10);
    }

    @Test
    public void testCourseEnrollWhenNoConflict(){
        ArrayList<Course> courses = new ArrayList<>();
        courses.add(curriculum.getCourseByCode("CSE1241"));
        student.enroll(courses,curriculum,parameters);
        assertEquals(courses.get(0).getName(), student.getEnrolledCourses().get(0));
    }

    @Test
    public void testCourseEnrollWhenTwoUpperSemester(){
        ArrayList<Course> courses = new ArrayList<>();
        courses.add(curriculum.getCOURSES()[2].get(0));
        student.enroll(courses,curriculum,parameters);
        assertEquals(0,student.getEnrolledCourses().size());
    }

    @Test
    public void testCourseEnrollWhenCollidingCourses(){
        Course course1 = curriculum.getCourseByCode("CSE1241");
        Course course2 = curriculum.getCourseByCode("MATH1001");
        ArrayList<Course> courses = new ArrayList<>();
        courses.add(course1);
        courses.add(course2);
        student.enroll(courses,curriculum,parameters);
        assertEquals(1, student.getEnrolledCourses().size());
        assertEquals(student.getEnrolledCourses().get(0), course1.getName());
    }

    @Test
    public void testCourseEnrollWhenMaxCourseLimitReached(){
        parameters.setMaxCoursePerSemester(1);
        Course course1 = curriculum.getCourseByCode("CSE1241");
        Course course2 = curriculum.getCourseByCode("MBG1201");
        ArrayList<Course> courses = new ArrayList<>();
        courses.add(course1);
        courses.add(course2);
        student.enroll(courses,curriculum,parameters);
        assertEquals(1, student.getEnrolledCourses().size());
        assertEquals(student.getEnrolledCourses().get(0), course1.getName());
    }

    @Test
    public void testCourseWhenPrerequisiteNotPassed(){
        Course course1 = curriculum.getCourseByCode("CSE1241");
        Course course2 = curriculum.getCourseByCode("CSE1242");
        ArrayList<Course> courses = new ArrayList<>();
        courses.add(course1);
        courses.add(course2);
        student.enroll(courses,curriculum,parameters);
        assertEquals(1, student.getEnrolledCourses().size());
        assertEquals(student.getEnrolledCourses().get(0), course1.getName());
    }
}

File: AdvisorTest.py
import unittest
from Curriculum import Curriculum
from JsonParser import JsonParser
from Student import Student


class AdvisorTest(unittest.TestCase):
    def setUp(self):
        from Advisor import Advisor
        self.advisors = [Advisor("advisor")]
        self.student = Student("Sanji", "1501210017", 1)
        self.student.advisor = self.advisors[0]
        self.parser = JsonParser()
        self.curriculum = Curriculum()
        try:
            self.parser.parseCourses(self.curriculum, self.advisors)
        except IOError as e:
            e.printStackTrace()
        self.parameters = self.parser.parseParameters()
        self.parameters.max_course_per_semester = 10

    def testCourseEnrollWhenNoConflict(self):
        courses = [self.curriculum.get_course_by_code("CSE1241")]
        self.student.enroll(courses, self.curriculum, self.parameters)
        self.assertEqual(courses[0].name, self.student.enrolled_courses[0])

    def testCourseEnrollWhenTwoUpperSemester(self):
        courses = [self.curriculum.courses[2][0]]
        self.student.enroll(courses, self.curriculum, self.parameters)
        self.assertEqual(0, len(self.student.enrolled_courses))

    def testCourseEnrollWhenCollidingCourses(self):
        course1 = self.curriculum.get_course_by_code("CSE1241")
        course2 = self.curriculum.get_course_by_code("MATH1001")
        courses = [course1, course2]
        self.student.enroll(courses, self.curriculum, self.parameters)
        self.assertEqual(1, len(self.student.enrolled_courses))
        self.assertEqual(self.student.enrolled_courses[0], course1.name)

    def testCourseEnrollWhenMaxCourseLimitReached(self):
        self.parameters.max_course_per_semester = 1
        course1 = self.curriculum.get_course_by_code("CSE1241")
        course2 = self.curriculum.get_course_by_code("MBG1201")
        courses = [course1, course2]
        self.student.enroll(courses, self.curriculum, self.parameters)
        self.assertEqual(1, len(self.student.enrolled_courses))
        self.assertEqual(self.student.enrolled_courses[0], course1.name)

    def testCourseWhenPrerequisiteNotPassed(self):
        course1 = self.curriculum.get_course_by_code("CSE1241")
        course2 = self.curriculum.get_course_by_code("CSE1242")
        courses = [course1, course2]
        self.student.enroll(courses, self.curriculum, self.parameters)
        self.assertEqual(1, len(self.student.enrolled_courses))
        self.assertEqual(self.student.enrolled_courses[0], course1.name)
